### PR TITLE
SDC: Allow Typing to filter for Dropdown component

### LINKED ISSUE #764: Allow typing to filter in the Dropdown widget if drop down has a large number of answer options
**Is your feature request related to a problem? Please describe.**
Currently the drop down widget has the keypad disabled, if we have a large number of answer options, we have to scroll through the entire set to find the answer. 

**Describe the solution you'd like**
If the drop down has a large number of answer options, we should enable to keypad to filter. This can also be decided based on the screen size. eg: Small screens can display a fewer number of answer options than large screens. So the number of answer options to allow filtering should not be a hard coded value but should be a function of the screen size.

### PR DESCRIPTION
**Pull request by nsabale7**
*IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #764 

**Description**
This PR allows to filter options based on typing in the dropdown component. 

**Type**
Choose one: Bug fix 

**Screenshots (if applicable)**

https://user-images.githubusercontent.com/82660329/201897060-6fca229a-8d9c-4661-b3b3-b83c610700a7.mp4

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

**Review by omarismail94 — COMMENTED** [review]
This looks great @nsabale7  !! 

Can you add an Espresso Test in `QuestionnaireItemDropDownViewHolderFactoryEspressoTest` to assert the number of items in the menu for different scenarios, e.g. no filtering, filtering with results, filtering with no results?

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDropDownViewHolderFactory.kt [hunk 1]
```diff
@@ -20,14 +20,14 @@ import android.content.Context
 import android.view.View
```
Comment (omarismail94) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDropDownViewHolderFactory.kt:20]: This looks great @nsabale7 !!

Can you add an Espresso Test in QuestionnaireItemDropDownViewHolderFactoryEspressoTest to assert the number of items in the menu for different scenarios, e.g. no filtering, filtering with results, filtering with no results?

Comment (omarismail94) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDropDownViewHolderFactory.kt:20]: I re-wrote the same comment as above to block the PR from being merged without resolution

Comment (nsabale7) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDropDownViewHolderFactory.kt:20]: @omarismail94 Done. Please have a look.

**Review by omarismail94 — APPROVED** [review]
Great change @nsabale7 !

**Comment by jingtang10** [comment]
@shelaghm please take a look at the video - it looks amazing!

thanks @nsabale7

**Comment by shelaghm** [comment]
@nsabale7 great work! 

@jingtang10 One question I have is if the dropdown component always allow people to type to filter or if it is something that is configurable and needs to be activated? Asking so I can document it properly in my design guidelines.

**Review by jingtang10 — APPROVED** [review]

**Comment by jingtang10** [comment]
> @nsabale7 great work!
> 
> @jingtang10 One question I have is if the dropdown component always allow people to type to filter or if it is something that is configurable and needs to be activated? Asking so I can document it properly in my design guidelines.

it would always allow filtering by typing.

**Pull request merged**